### PR TITLE
Javalab: Use codemirror

### PR DESCRIPTION
**Pull request by molly-moen**
Instead of using Ace as our editor in Javalab, switch to [Codemirror version 6](https://codemirror.net/6/)

Our main reason for the switch is codemirror 6 has been redesigned with accessibility in mind and has things such as tab navigation built in. Ace does not have a solution for tab navigation (you get stuck in the ace editor with no way out beyond us doing something hacky).

Note that this is a different version of codemirror than we use elsewhere because we want to take advantage of the new accessibility features. However the new codemirror has different package names than the old one so it should not be an issue.

We still allow tabbing in the editor, but the user can continue navigation by doing `esc`+`tab`.

The new editor looks like this:
Screen Shot 2021-02-18 at 4 23 04 PM

## Links

- [jira ticket](https://codedotorg.atlassian.net/browse/CSA-121)

## Testing story
Tested locally. Verified we are still writing updates to redux so we can save/run code later (we do not have that functionality yet).

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/javalab/JavalabEditor.jsx [hunk 1]
```diff
@@ -23,32 +25,41 @@ class JavalabEditor extends React.Component {
...
+      }),
+      parent: this._codeMirror,
+      dispatch: this.dispatchEditorChange()
+    });
   }
 
+  dispatchEditorChange = () => {
+    return tr => {
+      // we are overwriting the default dispatch method for codemirror,
+      // so we need to manually call the update method.
```
Comment (jmkulwik) [inline apps/src/javalab/JavalabEditor.jsx:41]: This commenting is very helpful :)

File: apps/src/javalab/JavalabEditor.jsx [hunk 2]
```diff
@@ -23,32 +25,41 @@ class JavalabEditor extends React.Component {
...
+          oneDarkHighlightStyle
+        ]
+      }),
+      parent: this._codeMirror,
+      dispatch: this.dispatchEditorChange()
+    });
   }
 
+  dispatchEditorChange = () => {
+    return tr => {
```
Comment (jmkulwik) [inline apps/src/javalab/JavalabEditor.jsx:39]: What is `tr` in this case?

**Comment by jmkulwik** [comment]
Haha I'm glad I'm in an empty room, because I might have otherwise startled anyone nearby. So excited about this change!!!

File: apps/src/javalab/JavalabEditor.jsx (hunk 2 above)
Comment (molly-moen) [reply apps/src/javalab/JavalabEditor.jsx:39]: tr is a codemirror [transaction](https://codemirror.net/6/docs/ref/#state.Transaction) I'll add a comment to make it more clear

Comment (jmkulwik) [reply apps/src/javalab/JavalabEditor.jsx:39]: 👍

File: apps/src/javalab/editorSetup.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+import {
```
Comment (jmkulwik) [inline apps/src/javalab/editorSetup.js:1]: This separation of concerns will make maintenance easy and straightforward 🙂

**Review by jmkulwik — COMMENTED** [review]
LGTM!

**Review by jmkulwik — COMMENTED** [review]
LGTM!

**Pull request merged**